garb/garb_main.cpp

```diff
@@ -13,6 +13,71 @@
 #include <thread>
 #include <chrono>
 
+#if defined(WITH_COREDUMPER) && WITH_COREDUMPER
+#include "coredumper/coredumper.h"
+
+#include <signal.h>
+
+void my_write_libcoredumper(int sig, const char *path, time_t curr_time) {
+  int ret = 0;
+  char suffix[512];
+  char core[512];
+  struct tm *timeinfo = gmtime(&curr_time);
+  if (path == NULL)
+    strcpy(core, "core");
+  else
+    strcpy(core, path);
+  sprintf(suffix, ".%d%02d%02d%02d%02d%02d", (1900 + timeinfo->tm_year),
+      timeinfo->tm_mon, timeinfo->tm_mday, timeinfo->tm_hour,
+      timeinfo->tm_min, timeinfo->tm_sec);
+  strcat(core, suffix);
+  fprintf(stderr, "CORE PATH: %s\n\n", core);
```

Comment:
I think using `fprintf` inside the signal handler is not safe
https://stackoverflow.com/questions/4554129/linux-glibc-can-i-use-fprintf-in-signal-handler. We should use `write` instead.